exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go

```diff
@@ -57,21 +49,39 @@ func TestNew(t *testing.T) {
 			},
 		},
 	}
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := New(tt.args.config, tt.args.httpAddress, tt.args.headers, tt.args.timeout)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("New() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-
-			if got == nil {
-				return
-			}
+			assert.NoError(t, err, "nil config")
+			assert.NotNil(t, got)
 
 			// This is expected to fail.
 			err = got.ConsumeTraceData(context.Background(), consumerdata.TraceData{})
 			assert.Error(t, err)
 		})
 	}
 }
+
+func TestNewFails(t *testing.T) {
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "empty_exporterName",
+			args: args{
+				config:      nil,
+				httpAddress: testHTTPAddress,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := New(tt.args.config, tt.args.httpAddress, tt.args.headers, tt.args.timeout)
+			assert.EqualError(t, err, "nil config")
```

Comment:
This assumes all test cases in `tests` (if we had more than one) will return the same error message. This may not be true. Generally unclear why we have `tests` array and loop through it if it is one element.